crates/sui/src/client_commands.rs

```diff
@@ -987,19 +1032,40 @@ impl SuiClientCommands {
                         skip_dependency_verification,
                     )
                     .await?;
-
-                let data = client
+                let tx_kind = client
                     .transaction_builder()
-                    .upgrade(
-                        sender,
+                    .upgrade_tx_kind(
                         package_id,
                         compiled_modules,
                         dependencies.published.into_values().collect(),
                         upgrade_capability,
                         upgrade_policy,
                         package_digest.to_vec(),
-                        gas,
+                    )
+                    .await?;
+                if dry_run {
+                    return helper_execute_dry_run(
```

Comment:
nit: we can omit `helper` and `execute` from the name here -- `helper` I don't think is adding much, and `execute` is a little confusing because that's how we refer to running a transaction for its effects.